// sanity/lib/_chunks-cjs/undeployAction.js
"use strict";
var _internal = require("./_internal.js"), helpers = require("./helpers.js");
const debug = _internal.debug.extend("undeploy");
async function undeployCoreAppAction(_, context) {
  const {
    apiClient,
    chalk,
    output,
    prompt,
    cliConfig
  } = context, client = apiClient({
    requireUser: !0,
    requireProject: !1
  }).withConfig({
    apiVersion: "v2024-08-01"
  });
  let spinner = output.spinner("Checking application info").start();
  const userApplication = await helpers.getUserApplication({
    client,
    appId: cliConfig && "__experimental_coreAppConfiguration" in cliConfig ? cliConfig.__experimental_coreAppConfiguration?.appId : void 0
  });
  if (spinner.succeed(), !userApplication) {
    output.print("Your project has not been assigned a Core application ID"), output.print("or you do not have __experimental_coreAppConfiguration set in sanity.cli.js or sanity.cli.ts."), output.print("Nothing to undeploy.");
    return;
  }
  output.print("");
  const url = `https://${chalk.yellow(userApplication.appHost)}.sanity.studio`;
  if (await prompt.single({
    type: "confirm",
    default: !1,
    message: `This will undeploy ${url} and make it unavailable for your users.
  The hostname will be available for anyone to claim.
  Are you ${chalk.red("sure")} you want to undeploy?`.trim()
  })) {
    spinner = output.spinner("Undeploying application").start();
    try {
      await helpers.deleteUserApplication({
        client,
        applicationId: userApplication.id,
        appType: "coreApp"
      }), spinner.succeed();
    } catch (err) {
      throw spinner.fail(), debug("Error undeploying application", err), err;
    }
    output.print(`Application undeploy scheduled. It might take a few minutes before ${url} is unavailable.`);
  }
}
exports.default = undeployCoreAppAction;
